src/project-tests/drum-machine-tests.js

```diff
@@ -12,20 +12,6 @@ export default function createDrumMachineTests() {
 
     // functions:
 
-    // As of React 15.6, we need a workaround that allows continued use of
-    // el.dispatchEvent()
-    // SEE: https://github.com/facebook/react/issues/10135
-    // Make <audio> Object configurable so that we can programmatically hit play
-    function configureAudioElement() {
```

Comment:
Have you confirmed that this isn't necessary?

Reply:
I read the mentioned issue and did not find any relation to our case. In addition, I checked, and it works with versions of React 15.3, 15.6 and 16.4.

Reply:
@ValeraS It was added to help with certain issues that came up with student projects -- not with the official projects. See #307 and #223  
It had to do with Object writability in React needing to be configured before certain elements could be programatically modified.

Reply:
@tbushman @223 is not related to this case. @307 is about mouse events. But here strange thing is done.  Because of this `paused` always is `undefined`

Reply:
@tbushman rebased

Reply:
@ValeraS also squash please

Reply:
@tbushman Why do we squash commits in the first place? IMO "Commit" should help to understand what was done to solve an individual task. If task small and we have made some commits only to correct a solution of the task (in a review process for example), then they should be squshed.

In this particular case, the commits solve different tasks, and if we will squash them, then we'll lose information about which changes were made for which task. Let's consider this commits as two different PR and let's merge them as is.

Reply:
I wish I was so sure. We've been ensuring each merge is squashed. 
@tchaffee ?

Reply:
That's how you personally organize your work locally (I do the same) and it's not that interesting when I look at an issue that was fixed six months ago. What I want is one commit that fixes the issue rather than having to piece together many commits that each fix a small part of the issue.

If a commit is big enough and important enough that it can stand on its own, then it deserves its own PR.

There is no problem with leaving multiple commits on a PR *while you are working on it*. That's fine. But before merge they should all be squashed. Almost every mature project on github requires it.

More info here.

https://softwareengineering.stackexchange.com/questions/263164/why-squash-git-commits-for-pull-requests#263172

Reply:
@tchaffee It's my fault what I've added the second commit in this PR. But this is not the case that you are about. Each of this commits solves different issues. Because of this, I suggest merging as is.

Reply:
If each commit solves different issues, please open another PR. We don't intentionally merge PRs with more than one commit.

Reply:
@ValeraS Our workflow is open for discussion for sure, but now isn't a great time with almost 50 open issues. I'm not even the one who would make the decision. We would have to get Quincy, Mrugresh, and probably Stuart involved in that kind of decision. And they are all extremely busy right now trying to get the new platform stable so I try to only get them involved if something is an obvious and big improvement and we need them to make a decision. For now I would suggest that you just follow the existing workflow and rules the various freeCodeCamp repos and if you find that after fixing a bunch of issues you still have problems with the workflow, then feel free to open an issue and we will get everyone involved and discuss it. For sure if you're going to open an issue to change the workflow be prepared to explain what's broken with our existing workflow, what problems it is causing, and how the new workflow would solve those problems. Thanks for your understanding.

Reply:
@tchaffee Formalism is not always good. You need to be more flexible.

You are saying it's not a great time for discussion with almost 50 open issues, and suggest to do unnecessary work in this particular case. How will it help resolve issues?

Reply:
@ValeraS Feel free to open a new issue if you want to change the existing workflow. I am working on the oldest issues first and we'll eventually get to your issue about changes to the workflow. In the meantime, this workflow has worked well for many experienced devs, the other freeCodeCamp repos, and most of the major repos on github. So we'll stick with it for now until we can get some of the much older issues fixed and then we will definitely make time for that discussion.

Reply:
@tchaffee I'm not asking you to change the rules, I'm just saying that any rule has exceptions.

Reply:
@tchaffee If I create two separate PRs, you will get the same result with the same commits after the merge. Can you explain to me how this will make our history cleaner?

Reply:
@ValeraS The choice is two PRs or squashed commits and either is fine. For the purpose of keeping track of which PRs trigger which issue closures, and if you're interested in distinguishing between these commits, the two separate PRs would each close an associated issue automatically on merge. In my experience it seems like only the first listed issue in the PR message is automatically closed on merge.

Reply:
@ValeraS You just submitted a PR about a week ago that will "force" users to follow a very specific format for commit messages. And yet you're now resisting following this rule.... Why would you create a PR that forces users to a specific commit message format if you at the same time want to argue about our existing workflow and rules? If you think the workflow and the rules are wrong, please open an issue and we will discuss in that issue when we get to it. Until we can make time for your suggestions, the rules are the rules. Thanks for your cooperation.

Reply:
@ValeraS As I said I'll be happy to discuss changing our workflow in another issue and when we get to it. In the meantime, if you would like your PR merged, you'll have to be flexible and go along with the existing workflow. Your choice.